pkg/util/mount/mount_helper_windows.go

```diff
@@ -0,0 +1,59 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+import (
+	"os"
+	"syscall"
+
+	"k8s.io/klog"
+)
+
+const (
+	// failure code is from https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes--1300-1699-
+	ERROR_LOGON_FAILURE = 1326
+)
+
+// IsCorruptedMnt return true if err is about corrupted mount point
+func IsCorruptedMnt(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var underlyingError error
+	switch pe := err.(type) {
+	case nil:
+		return false
+	case *os.PathError:
+		underlyingError = pe.Err
+	case *os.LinkError:
+		underlyingError = pe.Err
+	case *os.SyscallError:
+		underlyingError = pe.Err
+	}
+
+	if ee, ok := underlyingError.(syscall.Errno); ok {
+		if int(ee) == ERROR_LOGON_FAILURE {
```

Comment:
thank you for the code update. My main issue is why are we only checking for logon failure on windows and not the other network related issues that the linux codebase is checking for? Patrick had given 3 more network related failures to account for as well.

Reply:
@michmike thanks, I have added those 3 network failures in the code.